Use type annotations to constrain integer solutions.

joelburget opened the pull request:
In the case a dataclass member is annotated as an int, we can tell sympy
to generate only integer solutions for it. For now, we assume anything
that's not an int is a float.

Commit:
Use type annotations to constrain integer solutions.

In the case a dataclass member is annotated as an int, we can tell sympy
to generate only integer solutions for it. For now, we assume anything
that's not an int is a float.

Fixes #7

joelburget commented:
Hi Daniel, a couple of notes about this change:

* You'll notice I removed `solve`, since it's not used anymore.
* Actually, `System` is also no longer used by the wrapper. It wasn't clear to me if the `System` class has some capability that my use of sympy doesn't. If so we should add a test for that functionality.

Please let me know if I got either of these wrong.

DanielSank commented on constraintula/core.py:
```diff
@@ -210,12 +212,22 @@ def get_functions(self) -> Mapping[Symbol, Callable]:
-    assignments = {Symbol(k): v for k, v in independents.items()}
-    values = System(constraints).with_independents(assignments).evaluate(assignments)
-    return {k: values[Symbol(k)] for k in variables}
+def collect_symbols(expr):
```
Type annotations?

joelburget replied:
I added annotations but I wasn't sure how to check them. What command do you use?

DanielSank replied:
Well, the simplest thing is to pip install mypy and just run it against the target .py file. We could set up a script to run against all source files.

DanielSank reviewed (commented):
This looks really nice. Thanks! I have only one small nitpick.

Commit:
Add type annotations.

Commit:
Check whether symbol name is actually in args.

This is not the case in the circle test where the set of symbols would
be `{radius}`, since that's the only declared argument.
